src/trans.c

```diff
@@ -1369,28 +1369,34 @@ Obj FuncSMALLEST_IDEM_POW_TRANS (Obj self, Obj f) {
 *******************************************************************************/
 
 // Returns True if the transformation or list <t> is injective on the list <l>.
-// This has the minimal checks on arguments for performance reasons.
-
-// FIXME this should have more checks, since the GAP functions that call this
-// don't have checks.
-// TODO and then rename this IsInjectiveListTrans and remove the functions of
-// this name in lib/trans.gi
 
 Obj FuncIsInjectiveListTrans (Obj self, Obj l, Obj t) {
   UInt    n, i, j;
   UInt2   *ptt2;
   UInt4   *pttmp = 0L;
   UInt4   *ptt4;
+  Obj     val;
 
+  if (!IS_LIST(l)) {
+      ErrorQuit("the first argument must be a list but found %s",
```

Comment:
Sure, that's something annoying we show in lots of place right now, though. If it bothers you, we could start an effort to correctly print a/an by providing a function that returns the TNAM string, together with `a` or `an` correctly prefixed.

Or else, a PR that changes `(not a %s)` everywhere.

But I am not so happy about mixing this...